packages/ann/ann/c_src/logistic_actf_component.cc

```diff
@@ -58,4 +58,63 @@ namespace ANN {
     buffer.printf("ann.components.actf.logistic{ name='%s' }", name.c_str());
     return buffer.to_string(buffer_list::NULL_TERMINATED);
   }
+
+  SparseLogisticActfANNComponent::SparseLogisticActfANNComponent(const char *name,
+        float beta, float avg_act):LogisticActfANNComponent(name) {
+      this->beta = beta; this->avg_act = avg_act; } 
+
+  ANNComponent *SparseLogisticActfANNComponent::clone() {
+    SparseLogisticActfANNComponent *obj = new SparseLogisticActfANNComponent(name.c_str(), beta, avg_act);
+    return obj;
+  }
+
+  char *SparseLogisticActfANNComponent::toLuaString() {
+    buffer_list buffer;
+    buffer.printf("ann.components.actf.sparse_logistic{ name='%s', beta=%f, rho=%f  }", name.c_str(), this->beta, this->avg_act);
+    return buffer.to_string(buffer_list::NULL_TERMINATED);
+  }
+ 
+  void SparseLogisticActfANNComponent::multiplyDerivatives(MatrixFloat *input_units,
+						     MatrixFloat *output_units,
+						     MatrixFloat *input_errors,
+						     MatrixFloat *output_errors) {
+    UNUSED_VARIABLE(input_units);
+    Kernels::applyLogisticDerivative(output_errors, output_units);
```

Comment:
I'll move it to the end. So we could change it for the bp of the logistic function.